akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -434,12 +412,12 @@ class TLSActor(
     if (tracing) log.debug("handshake finished")
     val session = engine.getSession
 
-    hostInfo.map(_._1) match {
-      case Some(hostname) if !hostnameVerifier.verify(hostname, session) ⇒
```

Comment:
If I remember correctly this was because we supported Java 6 back then, and yes currently we can rely on `setEndpointIdentificationAlgorithm` so good to clean this up 👍